docs/articles/documentation/guides/advanced-guides/inject-client-scripts.md

```diff
@@ -0,0 +1,254 @@
+
+### Inject a JavaScript File
+
+Specify the JavaScript file path to inject the content of this file into the tested pages. You can pass a string or object with the `path` property.
+
+```sh
+testcafe chrome my-tests --cs assets/jquery.js
+```
+
+```js
+runner.clientScripts('assets/jquery.js');
+```
+
+```js
+{
+    "clientScripts": "assets/jquery.js"
+}
+```
+
+See the details for:
+
+* [command line interface](../../reference/command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2),
+* [programming interface](../../reference/api/runner/clientscripts.md#inject-a-javascript-file),
+* [configuration file](../../reference/configuration-file.md#inject-a-javascript-file),
+* test API: [fixture.clientScripts](../../reference/test-api/fixture/clientscripts.md#inject-a-javascript-file), [test.clientScripts](../reference/test-api/test/clientscripts.md#inject-a-javascript-file).
+
+### Inject a Module
+
+Specify the Node.js module's name to inject its content into the tested pages. Use an object with the `module` property.
+
+```js
+fixture `My fixture`
+    .page `https://example.com`
+    .clientScripts({ module: 'lodash' });
+```
+
+```js
+{
+    "clientScripts": {
+        "module": "lodash"
+    }
+}
+```
+
+TestCafe uses Node.js mechanisms to search for the module's entry point and injects its content into the tested page.
+
+Note that the browser must be able to execute the injected module. For example, modules that implement the [UMD](https://github.com/umdjs/umd) API can run in most modern browsers.
+
+See the details for:
+
+* [programming interface](../../reference/api/runner/clientscripts.md#inject-a-module),
+* [configuration file](../../reference/configuration-file.md#inject-a-module),
+* test API: [fixture.clientScripts](../../reference/test-api/fixture/clientscripts.md#inject-a-module), [test.clientScripts](../reference/test-api/test/clientscripts.md#inject-a-module).
+
+### Inject Script Code
+
+You can pass an object with the `content` property to provide the injected script as a string.
+
+```js
+const mockDate = `
+    Date.prototype.getTime = function () {
+        return 42;
+    };
+`;
+
+test
+    ('My test', async t => { /* ... */ })
+    .clientScripts({ content: mockDate });
+```
+
+```json
+{
+    "clientScripts": {
+        "content": "Date.prototype.getTime = () => 42;"
+    }
+}
+```
+
+See the details for:
+
+* [programming interface](../../reference/api/runner/clientscripts.md#inject-script-code),
+* [configuration file](../../reference/configuration-file.md#inject-script-code),
+* test API: [fixture.clientScripts](../../reference/test-api/fixture/clientscripts.md#inject-script-code), [test.clientScripts](../reference/test-api/test/clientscripts.md#inject-script-code).
+
+### Provide Scripts for Specific Pages
+
+You can also specify pages into which a script should be injected. For instance, this enables you to mock browser API on particular pages and use the default behavior everywhere else.
```

Comment:
```suggestion
You can also specify pages into which a script should be injected. This will allow you to mock browser API on specified pages and use the default behavior everywhere else.
```